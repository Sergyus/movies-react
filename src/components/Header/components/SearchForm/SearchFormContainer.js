import { connect } from 'react-redux';
import SearchForm from './SearchForm';
import { fetchSearchMovies, changeFilter } from '../../../../modules/app';

const actions = {
  fetchSearchMovies,
  changeFilter,
};

export default connect(null, actions)(SearchForm);
