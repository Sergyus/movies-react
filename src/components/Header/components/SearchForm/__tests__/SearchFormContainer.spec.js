import React from 'react';
import { create } from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import SearchFormContainer from '..';

describe('SearchFormContainer component renders correctly', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const initialState = {};
  const store = mockStore(initialState);
  it('SearchFormContainer renders correctly', () => {
    const tree = create(
      <Provider store={store}>
        <SearchFormContainer />
      </Provider>,
    )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
