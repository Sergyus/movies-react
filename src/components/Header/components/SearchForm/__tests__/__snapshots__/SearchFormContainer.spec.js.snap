// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFormContainer component renders correctly SearchFormContainer renders correctly 1`] = `
<form
  className="searchForm"
  onSubmit={[Function]}
>
  <input
    aria-label="Search Input"
    className="input"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="Search..."
    type="search"
    value=""
  />
  <button
    aria-label="Search Button"
    className="searchBtn"
    type="submit"
  />
</form>
`;
